File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/TempBattleStatBoosterModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.TempBattleStat;
import com.sfh.pokeRogueBot.model.modifier.ChooseModifierItem;
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class TempBattleStatBoosterModifierItem extends ModifierItem implements GeneratedPersistentModifierType {

    public static final String TARGET = "TempBattleStatBoosterModifierType";

    private TempBattleStat tempBattleStat;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/TempBattleStatBoosterModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.TempBattleStat
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType

class TempBattleStatBoosterModifierItem : ModifierItem(), GeneratedPersistentModifierType {

    companion object {
        const val TARGET = "TempBattleStatBoosterModifierType"
    }

    var tempBattleStat: TempBattleStat? = null
}